File: fvf4j/src/main/java/fvf4j/demo/domain/MerchantDirectory.java
package fvf4j.demo.domain;


import fvf4j.demo.domain.CategorizedTransaction.ExpenseCategory;
import fvf4j.demo.domain.Transaction.MerchantCategoryCode;

@FunctionalInterface
public interface MerchantDirectory {
    ExpenseCategory getFor(MerchantCategoryCode code);
}

File: fvf4k/src/main/kotlin/fvf4k/demo/domain/spi/MerchantDirectory.kt
package fvf4k.demo.domain.spi

import arrow.core.raise.context.Raise
import fvf4k.demo.domain.failure.Failure
import fvf4k.demo.domain.model.ExpenseCategory
import fvf4k.demo.domain.model.MerchantCategoryCode

fun interface MerchantDirectory {
    context(_: Raise<Failure>)
    fun getFor(mcc: MerchantCategoryCode): ExpenseCategory
}